File: mvx/src/main/java/com/yubin/mvx/databinding/dagger/AppInjector.java
package com.yubin.mvx.databinding.dagger;

import android.app.Application;

/**
 * <pre>
 *     @author : dingyubin
 *     time   : 2018-1-16
 *     desc   : 初始化
 *     version: 1.0
 * </pre>
 */

public class AppInjector {
    private static AppComponent appComponent;

    private AppInjector() {
        //no instance
    }

    public static void init(Application app) {
//        DaggerAppCompaonent
        appComponent.inject(app);
    }
}

File: mvx/src/main/java/com/yubin/mvx/databinding/dagger/AppInjector.kt
package com.yubin.mvx.databinding.dagger

import android.app.Application

/**
 * <pre>
 * @author : dingyubin
 * time   : 2018-1-16
 * desc   : 初始化
 * version: 1.0
</pre> *
 */
object AppInjector {

    private lateinit var appComponent: AppComponent

    fun init(app: Application) {
        appComponent = DaggerAppComponent.builder().application(app).build()

        appComponent.inject(app)
    }
}